src/HsmEventDispatcherSTD.cpp

```diff
@@ -200,8 +200,9 @@ void HsmEventDispatcherSTD::handleTimers() {
                 if (itTimeout != mRunningTimers.end()) {
                     if (true == handleTimerEvent(waitingTimerId)) {
                         // restart timer
+                        auto timer_interval = itTimeout->second.elapseAfter - itTimeout->second.startedAt;
```

Comment:
This solves original issue, but creates a new one. Due to thread scheduling, timers can be started with a delay. To mitigate this I had to calculate actual timer offset based on current time + timer interval. Calculating timer_interval in this way would in some cases introduce invalid timer duration (1~2 ms less than it's supposed to be).